File: pages/refactorings/replace-mutable-fields-with-lenses/03-file-tree/Before.hs
-- Grow the size of every node of a file tree by the same amount.
module Before where

data Node = Node { name :: String, size :: Int, children :: [Node] }
  deriving (Eq, Show)

bumpSizes :: Node -> Int -> Node
bumpSizes n by =
  n { size = size n + by
    , children = map (`bumpSizes` by) (children n)
    }

File: pages/refactorings/replace-mutable-fields-with-lenses/03-file-tree/Before.scala
// Grow the size of every node of a file tree by the same amount.
object Before:
  case class Node(name: String, size: Int, children: List[Node])

  def bumpSizes(n: Node, by: Int): Node =
    n.copy(
      size = n.size + by,
      children = n.children.map(bumpSizes(_, by)))
